### PR TITLE
PHPC-1944: Add final modifier to __set_state methods

### PR DESCRIPTION
**Pull request by levon80999**
https://jira.mongodb.org/browse/PHPC-1944

### CODE REVIEW

File: src/BSON/ObjectId.stub.php [hunk 1]
```diff
@@ -15,7 +15,7 @@ public final function getTimestamp(): int {}
 
     public final function __toString(): string {}
 
-    public static function __set_state(array $properties): ObjectId {}
+    public final static function __set_state(array $properties): ObjectId {}
```
Comment (jmikola) [inline src/BSON/ObjectId.stub.php:18]: For consistency, please place the `final` keyword _before_ the method visibility (e.g. `final public static`). I realize many of the stubs are inconsistent about this. I've opened [PHPC-2136](https://jira.mongodb.org/browse/PHPC-2136) to address that.

**Review by jmikola — CHANGES_REQUESTED** [review]
`configure~` should not have been included in this PR. Please remove that file.

Additionally, this PR only updates `ObjectId::__set_state()`. There are many more `__set_state()` methods in the extension that also need to be updated.

File: src/BSON/ObjectId.stub.php (hunk 1 above)
Comment (levon80999) [reply src/BSON/ObjectId.stub.php]: I've removed configure~ and add final to other __set_state() methods and by the way moving final keyword before the method visibility where I noted it but still there are many places where order is not correct

Comment (levon80999) [reply src/BSON/ObjectId.stub.php]: Also after generation stubs I noticed that order for key words does not any affect to generated files.

this is the code fragment that add arg infos
`private function getFlagsAsArginfoString(): string
    {
        $flags = "ZEND_ACC_PUBLIC";
        if ($this->flags & Class_::MODIFIER_PROTECTED) {
            $flags = "ZEND_ACC_PROTECTED";
        } elseif ($this->flags & Class_::MODIFIER_PRIVATE) {
            $flags = "ZEND_ACC_PRIVATE";
        }

        if ($this->flags & Class_::MODIFIER_STATIC) {
            $flags .= "|ZEND_ACC_STATIC";
        }

        if ($this->flags & Class_::MODIFIER_FINAL) {
            $flags .= "|ZEND_ACC_FINAL";
        }

        if ($this->flags & Class_::MODIFIER_ABSTRACT) {
            $flags .= "|ZEND_ACC_ABSTRACT";
        }

        if ($this->isDeprecated) {
            $flags .= "|ZEND_ACC_DEPRECATED";
        }

        return $flags;
    }`

Comment (jmikola) [reply src/BSON/ObjectId.stub.php]: > there are many places where order is not correct

Indeed, this is why I opened [PHPC-2136](https://jira.mongodb.org/browse/PHPC-2136). That can be addressed independently from this PR.

The order of modifiers has no impact on the generated C code or evaluation by PHP. I just want this addressed so the prototypes throughout our stubs are consistent.

Comment (levon80999) [reply src/BSON/ObjectId.stub.php]: If I understand correct there is no more changes. Can we merge this PR and continue work on PHPC-2136 after next triage

File: src/BSON/Binary.stub.php [hunk 2]
```diff
@@ -63,13 +63,13 @@ final class Binary implements BinaryInterface, \JsonSerializable, Type, \Seriali
      */
     public const TYPE_USER_DEFINED = UNKNOWN;
 
-    public final function __construct(string $data, int $type) {}
+    final public function __construct(string $data, int $type) {}
```
Comment (jmikola) [inline src/BSON/Binary.stub.php:66]: This PR should only change `__set_state` methods. Please leave these other corrections to [PHPC-2136](https://jira.mongodb.org/browse/PHPC-2136) (as mentioned in https://github.com/mongodb/mongo-php-driver/pull/1357#discussion_r967541061) so we don't end up with unrelated changes in the final commit.

**Review by jmikola — CHANGES_REQUESTED** [review]

File: src/BSON/Binary.stub.php (hunk 2 above)
Comment (levon80999) [reply src/BSON/Binary.stub.php]: Done, please review again

Comment (levon80999) [reply src/BSON/Binary.stub.php]: There is one CI fail I will fix it and request review again

Comment (levon80999) [reply src/BSON/Binary.stub.php]: Done

**Review by jmikola — APPROVED** [review]
